dagmc/dagnav.py

```diff
@@ -165,6 +165,9 @@ def __init__(self, model: DAGModel, handle: np.uint64):
         self.model = model
         self.handle = handle
 
+    def __del__(self):
+        self.model.used_ids[type(self)].discard(self.id)
```

Comment:
This is failing because by the time this line is being called, `self` has be reverted to `NoneType` as part of the destructor/cleanup process.  Does anyone know if there is a way to know the type of an object within the `__del__()` method? @paulromano @pshriwise @eepeterson

Reply:
Hmmm.... good question. In general, it would say that it's bad practice to associate a single MOAB EntitySet handle with multiple pydagmc sets, but of course it may happen.

I proposed this as a mechanism to avoid a polluted ID space that never frees up IDs, but perhaps that comes with more hassle than it's worth?

Reply:
Yep yep, good point @paulromano. I think the only place we should free IDs is in the `DAGSet.delete` method where the IDs are also freed in the underlying MOAB instance.